src/app_task.cpp

```diff
@@ -356,3 +386,67 @@ void AppTask::DispatchEvent(const AppEvent &event)
 		LOG_INF("Event received with no handler. Dropping event.");
 	}
 }
+
+void AppTask::UpdateTemperatureClusterState(const AppEvent &)
+{
+	static constexpr int16_t kTemperatureMeasurementAttributeMaxValue = 0x7fff;
+	static constexpr int16_t kTemperatureMeasurementAttributeMinValue = 0x954d;
+	static constexpr int16_t kTemperatureMeasurementAttributeInvalidValue = 0x8000;
+	static constexpr uint8_t kTemperatureMeasurementEndpointId = 1;
+
+	struct sensor_value sTemperature;
+	EmberAfStatus status;
+	int err = sensor_sample_fetch(sTemperatureSensor);
+	if (!err) {
+		err = sensor_channel_get(sTemperatureSensor, SENSOR_CHAN_DIE_TEMP, &sTemperature);
+		if (!err) {
+			int16_t newValue = static_cast<int16_t>(sTemperature.val1 * 100 +
+								sTemperature.val2 / 10000);
+
+			if (newValue > kTemperatureMeasurementAttributeMaxValue ||
+			    newValue < kTemperatureMeasurementAttributeMinValue) {
+				/* Read value exceeds permitted limits, so assign invalid value code to it. */
+				newValue = kTemperatureMeasurementAttributeInvalidValue;
+			}
+
+			LOG_INF("Current temperature: %d", newValue);
+
+			status = Clusters::TemperatureMeasurement::Attributes::MeasuredValue::Set(
+				kTemperatureMeasurementEndpointId, newValue);
+			if (status != EMBER_ZCL_STATUS_SUCCESS) {
+				LOG_ERR("Updating temperature measurement failed %x", status);
+			}
+		} else {
+			LOG_ERR("Getting temperature measurement data from sensor failed with: %d",
+				err);
+		}
+
+	} else {
+		LOG_ERR("Fetching data from temperature sensor failed with: %d", err);
+	}
+}
+
+void AppTask::ControlSensorHandler(const AppEvent &event)
+{
+	if (event.ButtonEvent.Action == static_cast<uint8_t>(AppEventType::ButtonPushed)) {
+		sSensorActive = !sSensorActive;
+		if (sSensorActive) {
+			ChipLogDetail(DeviceLayer, "Starting timer");
+			k_timer_start(&sSensorTimer, K_SECONDS(2), K_SECONDS(2));
+		} else {
+			ChipLogDetail(DeviceLayer, "Stopping timer");
+			k_timer_stop(&sSensorTimer);
+		}
+
+		DeviceLayer::SystemLayer().ScheduleLambda([] {
```

Comment:
I don't think On/Off cluster is needed at all. If you don't use it to control the sensor remotely, it's probably not. Also, in this version of the webinar we may implement a proper temperature sensor device type which does not include the OnOff cluster. It does include the Identify cluster so make sure you add the code for handling Identify commands.

Reply:
Removed. Identify implemented instead.